include/pipettes/core/i2c_writer.hpp

```diff
@@ -33,9 +33,11 @@ class I2CWriter {
 
     template <typename Data>
     requires std::is_integral_v<Data>
-    void write(Data data, uint16_t device_address) {
+    void write(Data data, uint16_t device_address, uint8_t reg = 0x0) {
```

Comment:
I think that rather than doing this at the level of the i2c writer, it's better to have some independent data formatters in a utility. Like you've noticed here, almost - _almost_ - every single i2c device has a register model where "read-register" transactions look like "WRITE regaddr + stop READ ", and write-register transactions look like "WRITE regaddr+continue DATA...", but not quite all of them do. So I think having some `namespace i2c_registers` that has functions like `auto build_register_write_message(uint8_t regaddr, std::ranges Data) -> MaxMessageBuffer;` and `auto build_register_read_message(uint8_t reg) -> MaxMessageBuffer` or whatever, and you can pass the result of those into i2c write.

Reply:
Or actually another good way to do this would be to have a second overload of `write`: `void write(uint16_t device_address, uint8_t reg, Data data)` that lives alongside this one